workspace_analyzer.py

```diff
@@ -851,228 +936,79 @@ def analyze_manipulability_on_object_surface(self,
-                    moller_results.append(async_startup[i].get(timeout=100))
-                print('Combining Results...')
-                for i in range(self.num_procs):
-                    #print(np.shape(moller_results[i]))
-                    num_intersections += moller_results[i]
-        else:
-            start = time.time()
-            for i in range(vec_len):
-                progressBar(i,
-                            vec_len - 1,
-                            prefix='Executing Moller Trumbore   ',
-                            ETA=start)
-                res = moller_trumbore_ray_intersection_array(
-                    verification_array, tri_vecs[i, :],
-                    np.array([EPSILON, 0, RAY_UNIT - EPSILON]))
+        #Step One: Filter Out All points Beyond Maximum Reach of the Arm
 
-                # where(res == True)
-                num_intersections[np.where(res)] += 1
+        #Step Two: Calculate Manipulability on All points within Maximum reach of the Arm
 
-        # Step Four: Discard Instances Where The Arm Collides With The Object In Question
-        lva = len(verification_array)
+        empty_results = []
+        true_points = []
         start = time.time()
-        penalized_configs = []
-        for i in range(lva):
+        for i in range(num_points):
             progressBar(i,
-                        lva - 1,
-                        prefix='Filtering out Collisions    ',
+                        num_points - 1,
+                        prefix='Preprocessing Data          ',
                         ETA=start)
-            if isodd(num_intersections[i]):
-                #sample_index = (i*true_rez*nd)+(j*nd)+k For reference
-                top_ind = i // (true_rez * nd)  # Top Level Bucket
-                snd_ind = i // (nd)  # Second Level Bucket
-                if exempt_ee and snd_ind == nd - 1:
-                    #TODO(Liam) is this correct?
-                    continue
-
-                config_duo = (top_ind, snd_ind)
-                if config_duo in penalized_configs:
+            p = points[i, :]
+            #Ignore points we've already filtered out
+            if bound_shape is not None:
+                if p not in filtered_points:
+                    empty_results.append(process_empty(p))
                     continue
-                penalized_configs.append(config_duo)
-                if sum(abs(results[top_ind][0])) < .001:
-                    #Is near zero
+            #Ignore points which are too far away
+            if fsr.distance(p, bot_base) > self.max_dist:
+                #TODO(Liam) determine max reach from AShape
+                empty_results.append(process_empty(p))
+                continue
+            #Ignore points which are too close together
+            if minimum_dist > 0:
+                continuance = False
+                for point in seen_points:
+                    if fsr.distance(point, p) < minimum_dist:
```

Comment:
4 levels of for/if/else... difficult to read and know what's going on. Consider helper functions or other means to improve readability of the code.

Reply:
Helper function used to reduce visible level to 3.